---
title: "PCA Homework"
author: "Thomas Gow"
date: "11/8/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# PCA Homework Cluster Outliers  

## The prompt:
The prompt is in the PCA folder named email.txt:

"Dear Data Team,

We need your help! Our labs currently have a contract with MIT researchers to measure gene expressions from 53 total samples of DNA for Leukemia research with our microarray technology. We have never been involved in analyzing the data from these types of experiments, our primary goal is merely to create the numerical data for the researchers. 

In this project, we were given 37 labeled samples from patients with 3 different types of Leukemia known before hand. The remaining 16 samples were not labeled by type.

Unfortunately, when our undergraduate interns finished the microarray processing, we found we had 38 labeled samples and 15 unlabeled samples. Nobody has any idea what happened, and now we have to figure out if one of these 38 labeled samples is incorrectly labeled. Since there are 3 types of Leukemia in both collections, there is a 2/3 chance the extra sample is labeled incorrectly.

Rather than profiling every single sample and matching it with an original (this is both expensive and extremely time consuming) we are hoping you can use some magical analytics to narrow in on a few specific samples that look unusual and might be tested first. 

The attached dataset contains the numerical microarray data - 5000 gene expression levels for each sample. These gene expressions are all measured on the same scale of expression, with a value of 20 representing no expression. You may find genes that look exactly the same across all samples (the 24th gene, for example, is highly unlikely to be expressed in any sample) The last column provides the type of Leukemia:

	1. AML (Acute Myeloid)
	2. ALL-B (Acute Lymphoblastic B-Cell)
	3. ALL-T (Acute Lymphoblastic T-Cell)

We need a response from you by end of business day September 15th at the latest because the profiling will take us two weeks and the project delivery date is scheduled for October 7th. If there is nothing you can do, we completely understand - this is just a shot in the dark.


Thank you,

Olive Peceay
Senior Researcher
Genomics Research Lab
EpiCenter Enterprises
(919) 867-5309 "


## Analysis
Loading in the data 
```{r data}
library(ggplot2)
# Load in R data table
load('/Users/thomasgow/Desktop/untitled folder 2/LeukError-2.RData')

# Markdown file acting up not recognizing loaded data
leuk_df <- leuk
leuklabels_df <- leuklabels
```

## Viewing the Data
There are 38 observations with 5001 variables. Three different typw of Leukemia: AML; ALL-B; ALL-T.

```{r viewing data}
#38 observations 5001 variables
table(leuk$V24)
#Eg: should see value 20 for almost all 38 obs.

leuk$V5001
# 3 type of Leukemia: AML; ALL-B; ALL-T
```

## PCA is needed
This graph shows PCA is needed to capture as much information as possible to find outliers.

```{r Bad Scatterplots}
# plot random points
randomColumns = sample(2:5001,2)
plot(leuk_df[,randomColumns],col = leuklabels$Class)
```

## PCA
We conducted an analysis to condense the information from the 5000 gene expressions into two principal components.
The scree plot below shows a good elbow at four. However, for visualization purposes we chose to use two PCs to start.
```{r Scatterplots}
#pca
pcaOut = prcomp(leuk[,1:5000],3, scale = F)

#plot screeplot 
plot(pcaOut$sdev^2, type='b', ylab = 'Eigenvalue (Variance of Each Component)', main = 'DNA Screeplot')
```

## Plotting PCA1 & PCA2
We plotted the 38 samples using one component as the x-axis and the second component as the y-axis. The three different colors represent the three Leukemia types present in the dataset.

When plotted, the 38 samples are divided into approximately three clusters, one for each Leukemia type.

At first look you can see a few obvious outliers such as the purple example in the top left. Now let's try to put numbers on the graph for better reference points.

```{r PCs}
#plot
plot(pcaOut$x[,1],pcaOut$x[,2],col = leuklabels$Class, 
     xlab = "Principal Component 1", 
     ylab = "Principal Component 2", 
     main = 'DNA Samples Projected into 2-dimensions')

```

## Final Graph
After applying numbers to the graph we recommend profiling the samples, 2, 10, 19, and 35, as their placement in relation to the clusters indicates they may be mislabeled.

```{r Better Graph}
# Better graph
plotdata = data.frame(
  PC1 = pcaOut$x[,1],
  PC2 = pcaOut$x[,2],
  Type = leuk$V5001
)

ggplot(plotdata, aes(x = PC1, y=PC2)) + geom_point(aes(colour = Type),size=3) + 
  scale_color_manual(values = c("ALL-B" = "red", "ALL-T" = "purple", "AML" = "green")) +
  labs(title='Genetic Samples Projected into 2-dimensions', x='Principal Component 1', y='Principal Component 2') +
  theme(legend.position = 'right')+
  geom_text(aes(label=as.numeric(rownames(plotdata))),hjust=.5, vjust=.5, size=4)
```

## Final Conclusion
Our investigations revealed that samples 2, 10, 19, and 35 require profiling to determine if they have been labeled correctly.
